app/controllers/DeclarationDetailController.scala

```diff
@@ -63,6 +66,20 @@ class DeclarationDetailController @Inject()(authenticate: IdentifierAction,
       }
     }
 
+  def displayDetails(ref: String,
+                     page: Option[Int]
+                    ): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
+    for {
+      accountOpt <- apiConnector.getCashAccount(request.eori)
+      result <- accountOpt match {
+        case Some(account) =>
+          val (from, to) = cashAccountUtils.transactionDateRange()
+          retrieveCashAccountTransactionAndDisplay(account, from, to, ref, page)
+        case None => Future.successful(NotFound(errorHandler.notFoundTemplate))
```

Comment:
do we need a _ default case?

Reply:
No, all the scenarios have been covered as part of Some and None